src/App.js

```diff
@@ -1,66 +1,106 @@
-import React from 'react';
-import TodoList from "./components/TodoComponents/TodoList"
-import TodoForm from "./components/TodoComponents/TodoForm"
+import React from "react";
+import TodoList from "./components/TodoComponents/TodoList";
+import TodoForm from "./components/TodoComponents/TodoForm";
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
-  
+
   state = {
     tasks: [
       {
-        task: 'Organize Garage',
+        phrase: "Organize Garage",
         id: 1528817077286,
-        completed: true
+        completed: false
       },
       {
-        task: 'Bake Cookies',
+        phrase: "Bake Cookies",
         id: 1528817084358,
         completed: false
       }
     ],
-    taskInput:""
-  }
-  
-  changeHandler = e =>{
-    e.preventDefault()
+    taskInput: ""
+  };
+
+  // Takes user input as a value, assigns it as the name 
+  changeHandler = e => {
+    e.preventDefault();
     this.setState({
       [e.target.name]: e.target.value
-    })
+    });
+  };
 
-  }
-  
+  // Adds new task - task is set to taskInput above, id is random, default completed is false
   addTask = e => {
-    e.preventDefault()
-    const newTask ={
-    task: this.state.taskInput,
-    id: Date.now(),
-    completed: false,
-  }
-  this.setState ({
-    tasks: [...this.state.tasks, newTask],
-    taskInput: ""
-  })
+    e.preventDefault();
+    const newTask = {
+      phrase: this.state.taskInput,
+      id: Date.now(),
+      completed: false
+    };
+   
+    this.setState({
+      tasks: [...this.state.tasks, newTask],
+      taskInput: ""
+    });
+  };
+
+  // clear button.  Whatever the state is, switch it
+  clearCompleted = e => {
+    e.preventDefault();
+    this.setState({
+      tasks: this.state.tasks.filter(task => !task.completed)
+    });
+  };
+
+
+  toggleTask = id => {
+    console.log(id)
+    this.setState(prevTask => {
+      return {
+        tasks: prevTask.tasks.map(tasks => {
+          if (tasks.id === id) {
+            tasks.completed = !tasks.completed;
+            return tasks;
+          }
+            else {
+              return tasks;
+            }
+          }
+        )
+      }
+    })
   }
```

Comment:
Your nesting needs to be cleaned up alot. A very easy way to get confused is to make your nesting too complex. In this case I would have nested it as such:

```js
toggleTask = id => {
	this.setState(prevTask => {

		return {
			tasks: prevTask.tasks.map(tasks => {
				if (tasks.id === id) {
					tasks.completed = !tasks.completed;         
					return tasks;
				} else { 
					return tasks; 
				}
			}) // Combines the closing of the .map `(` and `{`
		}
	}) // Combines the closing of the setState `(` and `{` 
}

|or|

toggleTask = id => {
	this.setState(prevTask => 
			({tasks: prevTask.tasks.map(tasks => tasks.id === id ? {...tasks, completed:!completed}: tasks)})
	)
}

```